### PR TITLE
ENH: Add an initial QListView with model support and example.

### PR DESCRIPTION
**Pull request by hhslepicka**
## Description

This PR adds an initial support for Qt items at Happi.
Those are optional.

## Motivation and Context

Many projects with UI are now looking into Happi and there is a motivation for Happi-aware elements.

## How Has This Been Tested?

Locally

## Where Has This Been Documented?

Not yet.

## Screenshots (if appropriate):
Screen Shot 2020-03-09 at 3 34 13 PM

Closes https://github.com/pcdshub/happi/issues/103

Attn. @ronpandolfi

### CODE REVIEW

File: happi/qt/model.py [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        The parent widget
+
+    client : Client
+        A happi.Client instance
+
+    kwargs : dict
+        Additional arguments to be passed to the QListView constructor.
+    """
+    def __init__(self, parent=None, client=None, **kwargs):
+        super(HappiDeviceListView, self).__init__(parent=parent,
```
Comment (klauer) [inline happi/qt/model.py]:
```suggestion
        super().__init__(parent=parent,
```

File: happi/qt/model.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        Returns
+        -------
+        happi.Client
+        """
+        return self._client
+
+    @client.setter
+    def client(self, client):
+        """
+        The client to use for search.
```
Comment (klauer) [inline happi/qt/model.py]: No need for docstring in the setter

File: happi/qt/model.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        """
+        Update the model with new data from the search.
+        """
+        def create_item(entry):
+            itm = QtGui.QStandardItem(entry.name)
+            itm.setData(entry)
+            itm.setFlags(itm.flags() & ~QtCore.Qt.ItemIsEditable)
+            return itm
+
+        items = []
```
Comment (klauer) [inline happi/qt/model.py]:
```suggestion
        items = [create_item(entry) for entry in self._happi_entries]
```

File: happi/qt/model.py [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+        def create_item(entry):
+            itm = QtGui.QStandardItem(entry.name)
+            itm.setData(entry)
+            itm.setFlags(itm.flags() & ~QtCore.Qt.ItemIsEditable)
+            return itm
+
+        items = [create_item(entry) for entry in self._happi_entries]
+        if not self._happi_entries:
+            return
+        for entry in self._happi_entries:
```
Comment (klauer) [inline happi/qt/model.py]: Need to remove these 2 lines, or `items` will be duplicated

Comment (hhslepicka) [reply happi/qt/model.py]: Addressed via https://github.com/pcdshub/happi/pull/105/commits/e3af86cb8def198fb1d95ec58290be9ec985c065

**Pull request merged**